navigation_and_routing/lib/src/app.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:bookstore/src/auth/auth.dart';
+import 'package:flutter/material.dart';
+
+import 'data/library.dart';
+import 'data/library_scope.dart';
+import 'routing/auth_guard.dart';
+import 'routing/delegate.dart';
+import 'routing/parser.dart';
+import 'routing/route_state.dart';
+
+class Bookstore extends StatefulWidget {
+  const Bookstore({Key? key}) : super(key: key);
+
+  @override
+  _BookstoreState createState() => _BookstoreState();
+}
+
+class _BookstoreState extends State<Bookstore> {
+  final auth = BookstoreAuth();
+  late final BookstoreRouteGuard guard;
+  late final RouteState routeState;
+  late final BookstoreRouterDelegate routerDelegate;
+
+  /// Configure the parser with all of the app's allowed path templates.
+  final routeParser = TemplateRouteParser([
+    '/signin',
+    '/books',
+    '/authors',
+    '/settings',
+    '/books/new',
+    '/books/all',
+    '/books/popular',
+    '/books/:bookId',
+    '/authors/:authorId',
+  ]);
+
+  final library = Library()
+    ..addBook('Left Hand of Darkness', 'Ursula K. Le Guin', true, true)
+    ..addBook('Too Like the Lightning', 'Ada Palmer', false, true)
+    ..addBook('Kindred', 'Octavia E. Butler', true, false)
+    ..addBook('The Lathe of Heaven', 'Ursula K. Le Guin', false, false);
+
+  @override
+  void initState() {
+    guard = BookstoreRouteGuard(auth: auth);
+    routeState = RouteState(routeParser);
+    routerDelegate = BookstoreRouterDelegate(routeState, auth, guard);
+    auth.addListener(() async {
```

Comment:
both routerDelegate and auth need to be disposed

Reply:
same for routeState

Reply:
good catch, fixed.